### ISSUE TITLE
Shell script does not pass args correctly

### ISSUE THREAD

**Issue by bert2** [codecrafters-io/build-your-own-grep#24]
The `your_grep.sh` is not passing arguments with spaces correctly to `stack`.

Suppose our app is just this little program:

```haskell
module Main where

import System.Environment

main :: IO ()
main = do
  args <- getArgs
  putStrLn $ show args
```

With the current `your_grep.sh` we'll get the following output:

```
> ./your_grep.sh -E "b r"
["-E","b","r"]
```

If we change `your_grep.sh` from...

```bash
exec stack run --silent -- $@
```

...to (note the [double quotes](https://tiswww.case.edu/php/chet/bash/bashref.html#Special-Parameters))...

```bash
exec stack run --silent -- "$@"
```

...then we get something more useful:

```
> ./your_grep.sh -E "b r"
["-E","b r"]
```

The current reference solution has a little [`unwords`](https://hackage.haskell.org/package/base-4.17.0.0/docs/Prelude.html#v:unwords) hack to deal with this after the fact. The above change would not cause the reference solution to fail, though. The `unwords` call would just be redundant.

@rohitpaulk what do you think?

**Comment by rohitpaulk** [comment]
@bert2 ah, yeah quoting makes sense! Want to open a PR? I'll also check to see if we're doing this wrongly for other languages.

**Comment by bert2** [comment]
All the other languages in the grep challenge use `"$@"` instead of `$@`. I didn't check any other challenges.

**Comment by bert2** [comment]
Closed by #26.
